app/controllers/shopping_lists_controller.rb

```diff
@@ -18,6 +18,16 @@ def edit
     redirect_to @shopping_list
   end
 
+  def make_primary
+    ShoppingList.update_all(primary: false)
```

Comment:
So this technique of doing an update_all works and is much more efficient than looping through all ShoppingList's setting or unsetting primary.

However `update_all` does not trigger any rails callbacks on models, or test any validations, or touch any timestamps. This could be considered a little bit unsafe. 

Another way I could think to do this which would be efficient and also call rails callbacks etc is:

```ruby
ShoppingList.find_by(primary: true).update(primary: false)
ShoppingList.find(params[:shopping_list_id]).update(primary: true) # < this is the same as the rest of your action
```

Reply:
This can sometimes be helpful to navigate various active record methods: http://www.davidverhasselt.com/set-attributes-in-activerecord/

Reply:
Thanks, very useful link!

In 5270d43, I've update to use `.find_by`, and also the more concise `.update` style that you wrote above.